### DISCUSSION TITLE
How to load background to a 3D model without hdr file?

### DISCUSSION THREAD

**Discussion by tekaIvanishvili** [sceneview/sceneview#447] [Q&A]
I'm using the io.github.sceneview:sceneview:2.0.4 Model viewer. When I set the environment and load an HDR file, everything functions perfectly. However, I'm facing difficulty changing the background texture when not loading the HDR file. While I managed to add light to ensure the object is displayed using the following code:
```
val sun = EntityManager.get().create()
LightManager.Builder(LightManager.Type.DIRECTIONAL)
    .build(sceneView.engine, sun)
sceneView.scene.addEntity(sun)
```
I couldn't find a way to set the blank background. In the SceneForm documentation sample, it's achieved simply by setting the background attribute:
```
<com.google.ar.sceneform.SceneView
    android:id="@+id/scene_view"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:background="@color/deep_teal"/>
```
I appreciate your help. Below is the code snippet for loading the model:
```
suspend fun loadModel(modelFile: String) {
    sceneView.cameraNode.apply {
        position = io.github.sceneview.math.Position(z = 4.0f)
    }
    val modelInstance = sceneView.modelLoader.createModelInstance(modelFile)
    val modelNode = ModelNode(
        modelInstance = modelInstance,
        scaleToUnits = 2.0f,
    )
    modelNode.scale = io.github.sceneview.math.Scale(0.1f)
    sceneView.addChildNode(modelNode)
}
```

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, you can set a solid color background instead of an HDR skybox:

```kotlin
SceneView(
    environment = rememberEnvironment(
        skyboxColor = Color(0xFF2196F3) // solid blue background
    )
) {
    ModelNode(modelInstance = model)
}
```

You can also use a transparent background (useful for overlaying 3D on other UI) or just skip the skybox while still loading an HDR for indirect lighting only.
